cmd/soroban-rpc/internal/methods/simulate_transaction.go

```diff
@@ -345,9 +345,11 @@ func NewSimulateTransactionHandler(logger *log.Entry, ledgerEntryReader db.Ledge
 				})
 
 			default:
+				rv := base64.StdEncoding.EncodeToString(result.Result)
+				auth := base64EncodeSlice(result.Auth)
 				results = append(results, SimulateHostFunctionResult{
-					ReturnValueXDR: base64.StdEncoding.EncodeToString(result.Result),
-					AuthXDR:        base64EncodeSlice(result.Auth),
+					ReturnValueXDR: &rv,
+					AuthXDR:        &auth,
```

Comment:
Is there a code path where we create a `SimulateHostFunctionResult{}` instance with a nil `ReturnValueXDR` field and/or a nil `AuthXDR` field? Looking through this code, I could not find one.

Reply:
Yes, in the `case FormatJSON` above this one, the `SimulateHostFunctionResult` only fills out the `ReturnValueJSON` and `AuthJSON` fields.